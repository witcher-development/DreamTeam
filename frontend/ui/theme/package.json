{
  "name": "@ui/theme",
  "version": "0.1.0",
  "license": "MIT",
  "main": "src/index.ts",
  "peerDependencies": {
    "react": "16.8.6"
  },
  "dependencies": {
    "emotion-theming": "^10.0.9"
  }
}
