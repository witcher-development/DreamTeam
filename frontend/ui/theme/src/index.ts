import { ThemeProvider } from './Provider'

export { ThemeProvider }
