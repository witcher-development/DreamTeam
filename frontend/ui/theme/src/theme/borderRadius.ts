import { ThemeBorderRadius } from './types'

const borderRadius: ThemeBorderRadius = [2, 4, 8]

borderRadius.s = borderRadius[0]
borderRadius.n = borderRadius[1]
borderRadius.m = borderRadius[2]

export default borderRadius
