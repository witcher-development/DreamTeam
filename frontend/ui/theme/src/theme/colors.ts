const colors = {
  black: '#000000',
  white: '#ffffff',
  webWhite: '#F5F8FA',
  blueHaze: '#BFC8DC',
  lightGray: '#8390AD',
  blueBayoux: '#4D5A78',
  ebony: '#293246',
  red: '#F15151',
  orange: '#FFA800',
}

export default colors
