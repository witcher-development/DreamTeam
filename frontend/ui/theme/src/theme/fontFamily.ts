const fontFamily = {
  sf: '"SF UI Text", sans-serif',
}

export default fontFamily
