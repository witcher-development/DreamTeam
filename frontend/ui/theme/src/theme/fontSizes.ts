import { ThemeFontSizes } from './types'

const fontSizes: ThemeFontSizes = [14, 16, 20, 24, 32, 40, 48]

fontSizes.s = fontSizes[0]
fontSizes.n = fontSizes[1]
fontSizes.m = fontSizes[2]
fontSizes.l = fontSizes[3]
fontSizes.xl = fontSizes[4]
fontSizes['2xl'] = fontSizes[5]
fontSizes['3xl'] = fontSizes[6]

export default fontSizes
