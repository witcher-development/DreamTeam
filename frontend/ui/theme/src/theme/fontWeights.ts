import { ThemeFontWeights } from './types'

const fontWeights: ThemeFontWeights = [300, 400, 500, 600]

fontWeights.light = fontWeights[0]
fontWeights.normal = fontWeights[1]
fontWeights.medium = fontWeights[2]
fontWeights.bold = fontWeights[3]

export default fontWeights
