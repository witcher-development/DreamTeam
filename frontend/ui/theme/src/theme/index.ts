import borderRadius from './borderRadius'
import fontFamily from './fontFamily'
import fontSizes from './fontSizes'
import fontWeights from './fontWeights'
import colors from './colors'
import lineHeights from './lineHeights'

export const theme = {
  borderRadius,
  fontFamily,
  fontSizes,
  fontWeights,
  colors,
  lineHeights,
}

export * from './types'
