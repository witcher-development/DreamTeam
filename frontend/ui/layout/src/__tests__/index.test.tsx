import React from 'react'
import serializer from 'jest-emotion'
import renderer from 'react-test-renderer'
import { ThemeProvider } from '@ui/theme'
import { Column, Layout } from '../index'

expect.addSnapshotSerializer(serializer)

describe('Layout', () => {
  describe('column snapshot', () => {
    it('should match latest column render snapshot', () => {
      const column = renderer
        .create(
          <ThemeProvider>
            <Column />
          </ThemeProvider>
        )
        .toJSON()

      expect(column).toMatchSnapshot()
    })

    it('should match latest column fill render snapshot', () => {
      const column = renderer
        .create(
          <ThemeProvider>
            <Column fill />
          </ThemeProvider>
        )
        .toJSON()

      expect(column).toMatchSnapshot()
    })
  })

  describe('layout snapshot', () => {
    it('should match latest layout render snapshot', () => {
      const layout = renderer
        .create(
          <ThemeProvider>
            <Layout
              justify='center'
              align='center'
              basis='xl'
              grow={1}
              shrink={1}
            />
          </ThemeProvider>
        )
        .toJSON()

      expect(layout).toMatchSnapshot()
    })
  })
})
