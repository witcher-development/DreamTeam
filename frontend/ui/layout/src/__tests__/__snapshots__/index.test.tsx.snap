// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout column snapshot should match latest column fill render snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
}

<div
  className="emotion-0"
/>
`;

exports[`Layout column snapshot should match latest column render snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

<div
  className="emotion-0"
/>
`;

exports[`Layout layout snapshot should match latest layout render snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-basis: 40px;
  -ms-flex-preferred-size: 40px;
  flex-basis: 40px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

<div
  className="emotion-0"
/>
`;
